### PR TITLE
[Modules] Updated Virtual Network Module to the latest API Version (2022-07-01)

### LINKED ISSUE #2785: [Failed pipeline] Network - VirtualNetworks
failed run: https://github.com/Azure/ResourceModules/actions/runs/4168125360

### PR DESCRIPTION
**Pull request by ahmadabdalla**
Closes #2785 

# Description

Updated Virtual Network Module to the latest API Version (2022-07-01)

| Pipeline |
| - |
|[Network - VirtualNetworks](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.virtualnetworks.yml) |

# Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: modules/Microsoft.Network/virtualNetworks/deploy.bicep [hunk 1]
```diff
@@ -19,6 +19,23 @@ param ddosProtectionPlanId string = ''
 @description('Optional. Virtual Network Peerings configurations.')
 param virtualNetworkPeerings array = []
 
+@description('Optional. Indicates if VM protection is enabled for all the subnets in the virtual network.')
+param enableVmProtection bool = false
```
Comment (AlexanderSehr) [inline modules/Microsoft.Network/virtualNetworks/deploy.bicep:23]: Should we set this to true by default? Sounds like a meaningful security setting

File: modules/Microsoft.Network/virtualNetworks/deploy.bicep [hunk 2]
```diff
@@ -19,6 +19,23 @@ param ddosProtectionPlanId string = ''
...
+
+@description('Optional. Indicates if encryption is enabled on virtual network and if VM without encryption is allowed in encrypted VNet. Requires the EnableVNetEncryption feature to be registered for the subscription and a supported region to use this property.')
+param vnetEncryption bool = false
```
Comment (AlexanderSehr) [inline modules/Microsoft.Network/virtualNetworks/deploy.bicep:26]: Same comment as for the VM protection

Comment (AlexanderSehr) [reply modules/Microsoft.Network/virtualNetworks/deploy.bicep:26]: But I guess it's a bit tricky if it must be enabled on the subscription

Comment (ahmadabdalla) [reply modules/Microsoft.Network/virtualNetworks/deploy.bicep:26]: Yes not an out of the box feature

File: modules/Microsoft.Network/virtualNetworks/deploy.bicep (hunk 1 above)
Comment (ahmadabdalla) [reply modules/Microsoft.Network/virtualNetworks/deploy.bicep:23]: Seeing this ==> https://github.com/Azure/azure-cli/issues/14059#issuecomment-667289923

I want to remove it now :D

Comment (AlexanderSehr) [reply modules/Microsoft.Network/virtualNetworks/deploy.bicep:23]: Agreed

Comment (ahmadabdalla) [reply modules/Microsoft.Network/virtualNetworks/deploy.bicep]: Updated, tested locally and regenerated doco

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**